### ISSUE TITLE
[sfml] Allow release-only build

### ISSUE THREAD

**Issue by mhandb** [microsoft/vcpkg#18150]
When doing a release only build with :
```
set(VCPKG_BUILD_TYPE release)
```
SFML will fail to install as it tries to copy files from the debug build.
Here is the cuplrit line in `ports/sfml/portfile.cmake`
```    
file(COPY ${CURRENT_PACKAGES_DIR}/debug/lib/sfml-main-d.lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link)
```

**Comment by NancyLi1013** [comment]
Hi @mhandb 
Thanks for posting this issue.

Vcpkg will build debug type and release type by default. If you wants to build only release or debug, please modify portfile.cmake manually or use `--overlay-ports` to build.

For overlay-ports, please see https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md.

**Comment by mhandb** [comment]
Thanks, I will go with the ports overlays.
I understand that by default, vcpkg builds both, but it would be nice to support `set(VCPKG_BUILD_TYPE release)`
that can be added thanks to `--overlay-triplets`.

**Comment by HaoxuanGuo** [comment]
Hi @mhandb , can you explain it in more detail?

I met the same problem that a library which symbol is more than 65535 can not be compile in DEBUG mode. But when I add `set(VCPKG_BUILD_TYPE release)` into the port file, it is failed in due to
```
Error: Building package geant4:x64-windows failed with: POST_BUILD_CHECKS_FAILED
```
because of the check for
```
Mismatching number of debug and release binaries.
```

When the DEBUG is blocked by `set(VCPKG_BUILD_TYPE release)`, there are no files to be generated and this problem occured.

**Comment by mhandb** [comment]
My use case is simpler, I do not need to deliver the debug build and want to speedup the ci pipelines for my project.

**Comment by HaoxuanGuo** [comment]
> My use case is simpler, I do not need to deliver the debug build and want to speedup the ci pipelines for my project.

Sorry, my explanation may unclear. I just wonder how to modify the port file to build release mode only (skip debug mode) without errors. If just add `set(VCPKG_BUILD_TYPE release)` to skip debug build, it will throw an error because the number of binaries mismatch. Is there any method to avoid (block) this check?

**Comment by sfhacker** [comment]
@mhandb Are you ok?
@HaoxuanGuo What have you met? [1]


[1] https://english-for-chinese.blogspot.com/2008/06/and-meet-problem.html

**Comment by mhandb** [comment]
@sfhacker Yes I am fine :-D, And you?

@HaoxuanGuo
To fix the port to make it build in release only, it depends on the ports itself.
For SFML I changed the ports so it checks if the file (.lib) exists before trying to copy it. Then if the debug build is not built.
It will not fail trying to copy it.

**Comment by NancyLi1013** [comment]
@mhandb 
Seems you have solved your problems, right? You can close this issue if it is not a problem for you now.

@HaoxuanGuo 
If you want to build only release, you can use the same way to build. But from your description, it seems you cannot build it successfully by default. So it may be related with the port itself. Your problem is different from this issue.

I also checked that `geant4` is not in vcpkg. So I'm not clear what your problem is.

You can open a new issue to track your problem if you need help from vcpkg.

Thanks.

**Comment by HaoxuanGuo** [comment]
@sfhacker , thank you for you correction.

@sfhacker , @NancyLi1013 . OK. I will show my problems in details below.

I tried to build [geant4](https://geant4.web.cern.ch/) in vcpkg.

But due to it is not in the port registry, I have to [add a new package from a zip file](https://github.com/microsoft/vcpkg/blob/master/docs/examples/packaging-zipfiles.md). This library is too large to compile in DEBUG (it has more than 65535 symbols). vcpkg build both in default that cause an error.

To fix this error, I tried modifying the port file to disable DEBUG compiling. Thus, I add `set(VCPKG_BUILD_TYPE release)` into the port file. It works but another problem occurred.

Although the DEBUG is disabled, vcpkg also checks the number of debug and release binaries. Obviously, there is no debug binary and the number mismatch. Another error which I am unable to fix throwed.

In a nutshell, I wonder how can I modify the port file generated to build only release (no debug) in default triplets.
